src/main/java/com/project/springbootproject/controller/ShoppingCartController.java

```diff
@@ -0,0 +1,65 @@
+package com.project.springbootproject.controller;
+
+import com.project.springbootproject.dto.cartitemdto.CartItemDto;
+import com.project.springbootproject.dto.cartitemdto.CartItemQuantityDto;
+import com.project.springbootproject.dto.shoppingcartdto.ShoppingCartDto;
+import com.project.springbootproject.model.User;
+import com.project.springbootproject.service.shoppingcart.ShoppingCartService;
+import io.swagger.v3.oas.annotations.Operation;
+import io.swagger.v3.oas.annotations.tags.Tag;
+import java.security.Principal;
+import lombok.RequiredArgsConstructor;
+import org.springframework.http.HttpStatus;
+import org.springframework.security.authentication.UsernamePasswordAuthenticationToken;
+import org.springframework.web.bind.annotation.DeleteMapping;
+import org.springframework.web.bind.annotation.GetMapping;
+import org.springframework.web.bind.annotation.PathVariable;
+import org.springframework.web.bind.annotation.PostMapping;
+import org.springframework.web.bind.annotation.PutMapping;
+import org.springframework.web.bind.annotation.RequestBody;
+import org.springframework.web.bind.annotation.RequestMapping;
+import org.springframework.web.bind.annotation.ResponseStatus;
+import org.springframework.web.bind.annotation.RestController;
+
+@Tag(name = "ShoppingCart controller",
+        description = "Endpoint for managing ShoppingCarts")
+@RestController
+@RequiredArgsConstructor
+@RequestMapping(value = "/api/cart")
```

Comment:
make api project wide url prefix
in properties: `server.servlet.context-path=/api`